phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexScrutinyTool.java

```diff
@@ -259,18 +266,20 @@ public Job createSubmittableJob(String schemaName, String indexTable, String dat
 
             // set CURRENT_SCN for our scan so that incoming writes don't throw off scrutiny
             configuration.set(PhoenixConfigurationUtil.CURRENT_SCN_VALUE, Long.toString(ts));
+            String tenantId = configuration.get(PhoenixRuntime.TENANT_ID_ATTRIB);
 
             // set the source table to either data or index table
             SourceTargetColumnNames columnNames =
                     SourceTable.DATA_TABLE_SOURCE.equals(sourceTable)
                             ? new SourceTargetColumnNames.DataSourceColNames(pdataTable,
-                                    pindexTable)
+                                    pindexTable, tenantId)
                             : new SourceTargetColumnNames.IndexSourceColNames(pdataTable,
-                                    pindexTable);
+                                    pindexTable, tenantId);
             String qSourceTable = columnNames.getQualifiedSourceTableName();
-            List<String> sourceColumnNames = columnNames.getSourceColNames();
-            List<String> sourceDynamicCols = columnNames.getSourceDynamicCols();
-            List<String> targetDynamicCols = columnNames.getTargetDynamicCols();
+
+            List<String> sourceColumnNames = columnNames.getSourceColNamesForSkipScan();
```

Comment:
Seems like we're "leaking" implementation details here...is there a reason why the job creator function needs to know the Scan's going to be a skip scan? Why should it need to change in the future if we decide to use a different Scan filter?